docs/policy-management.md

```diff
@@ -0,0 +1,78 @@
+
+
+## Implementation details
+
+#### Authorization/Authentication decisions
+The current implementation has insufficient authentication restrictions. Currently, the only requirement is that the 'Authorization' header is to be set to the webID of the "logged on" client. Proper procedures to authenticate this client are still to be implemented.
+
+#### Sanitization decisions
+Some endpoints allow new policies to be created, or existing policies to be modified. This introduces the possibility of invalid syntactic or semantic policies, hence a sanitization strategy is required. In the current implementation, only POST could introduce such problems. We provided the following basic checks:
+- Every defined rule must have a unique ID.
+- Every rule must have exactly one assigner.
+- Every assigner must match the authenticated client.
+
+Sanitization Limitations
+- It is possible to `POST` a policy with an ID that already exists, or with rules that reuse already existing IDs.
+- There are currently no checks to verify whether a client is sufficiently authorized to create or modify a policy/rule for a specific target.
+    * A client should not be in able to alter rights about a target it does not have access to.
+- There are plenty of other sanitization checks to be considered. 
+
+#### URI encodig decision
+Some operations require the client to specify a policy ID in the URL. Since policy ID's might contain reserved characters (e.g. `/`, `:`, ...), we have chosen to encode them with the builtin [`encodeURIComponent()` function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent). Using this method, reserved characters will be converted to their respective UTF-8 encodings.
+
+## Testing
+The current implementation is tested only by the script in `scripts\test-uma-ODRL-policy.ts`. This script tests every implemented endpoint with a designated flow. Since the script initiates with an empty storage, and there is no endpoint or other way to seed it, the first requests must test the POST endpoint. These tests are designed to ensure that the storage is filled. After the POST tests, the access endpoints can be tested. 
+
+## Problems
+- When you have a policy with multiple rules that have different assigners, DELETE on every rule of one assigner will succesfully delete the rule itself, but not the definition of the rule within the policy. This is due to the fact that you can currently only DELETE based on the ID of the rule/policy you want to delete, and you cannot delete the entire policy since other assigners depend on it. Currently, the only problem with this is filling space, since the quads defining deleted rules will not be returned in GET requests.
```

Comment:
This is related to how the abstraction of the storage works.: https://github.com/lennertdr/user-managed-access/blob/main/packages/ucp/src/storage/UCRulesStorage.ts

Maybe this needs a redesign

e.g. deleteRuleFromPolicy(ruleID:string, PolicyID:string), etc

Reply:
Maybe, we should make an issue in the uma repo